test/test_images/probe_helper/probe_helper.go

```diff
@@ -243,6 +247,35 @@ func (ph *ProbeHelper) forwardFromProbe(ctx context.Context, brokerClient cloude
 			default:
 				return logNACK(ctx, "Probe forwarding failed, unrecognized cloud storage probe subject: %v", event.Subject())
 			}
+		case CloudAuditLogsSourceProbeEventType:
+			switch event.Subject() {
+			case CloudAuditLogsSourceProbeCreateSubject:
+				channelID = event.ID() + "-" + event.Subject()
+				receiverChannel, err = receivedEvents.createReceiverChannel(channelID)
+				if err != nil {
+					return logNACK(ctx, "Probe forwarding failed, could not create receiver channel: %v", err)
+				}
+				defer receivedEvents.deleteReceiverChannel(channelID)
```

Comment:
The channel management logic can be moved before `switch event.Subject()`.

Reply:
Agreed, fixed. This cleans up the code a bit. It's important to remember that, e.g., the (not yet merged) CloudSchedulerSource probe won't be channel-based and so we'll have to remember not to leak a channel for that probe.